app/generator/content/schedule/part3/get-summary.js

```diff
@@ -0,0 +1,18 @@
+const getReductionSummary = require('./get-reduction-summary')
+const getTopUpSummary = require('./get-top-up-summary')
+const getReductionZeroSummary = require('./get-reduction-zero-summary')
+const getRecoverySummary = require('./get-recovery-summary')
+
+const getSummary = (schedule) => {
+  const remainingAmount = schedule.remainingAmount
+  switch (true) {
+    case (remainingAmount > 0):
+      return schedule.adjustment.adjustmentValue > 0 ? getTopUpSummary(schedule.adjustment) : getReductionSummary(schedule)
+    case (remainingAmount === 0):
+      return getReductionZeroSummary(schedule)
+    case (remainingAmount < 0):
+      return getRecoverySummary(schedule)
+  }

```

Comment:
No case when remainingAmount is undefined because schedule doesn't have a remainingAmount key

Reply:
remainingAmount can never be undefined, it only allows numeric, it does not allow any. There can only be 3 possible values of numeric   >0, <0 and =0, hence all possible scenario are taken care of. any other condition or logic flow will be redundant.